### ISSUE TITLE
[Accessibility]: Search context management misc keyboard bugs

### ISSUE THREAD

**Issue by limitedmage** [sourcegraph/sourcegraph#34726]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34191

### Problem description

- [ ] "Manage context" link in search contexts dropdown cannot be activated with keyboard (neither space nor enter work)
![image](https://user-images.githubusercontent.com/206864/166073816-f778e447-9049-4391-b083-054daf278dd5.png)

- [ ] ~Opening the "Code search" dropdown menu removes focus ring and does not focus the first menu item immediately; focused menu item does not match focus styles elsewhere in the application.~ Issue to fix: https://github.com/sourcegraph/sourcegraph/issues/36350

https://user-images.githubusercontent.com/206864/166074026-828965b5-b949-44c9-a4cc-fba4f885e466.mp4

### Expected behavior

Both issues should be fixed

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.
